<launch>
  <arg name="input/lanelet_map_bin" default="/map/vector_map" />
  <arg name="input/obstacle_pointcloud" default="/sensing/lidar/no_ground/pointcloud" />
  <arg name="input/reference_trajectory" default="/planning/scenario_planning/trajectory" />
  <arg name="input/predicted_trajectory" default="/control/trajectory_follower/predicted_trajectory" />
  <arg name="input/twist" default="/localization/twist" />
  <arg name="config_file" default="$(find-pkg-share obstacle_collision_checker)/config/obstacle_collision_checker.param.yaml" />
  <arg name="vehicle_param_file" />

  <node pkg="obstacle_collision_checker" exec="obstacle_collision_checker_node" name="obstacle_collision_checker_node" output="screen">
    <param from="$(var config_file)" />
    <param from="$(var vehicle_param_file)" />
    <remap from="input/lanelet_map_bin" to="$(var input/lanelet_map_bin)"/>
    <remap from="input/obstacle_pointcloud" to="$(var input/obstacle_pointcloud)"/>
    <remap from="input/reference_trajectory" to="$(var input/reference_trajectory)"/>
    <remap from="input/predicted_trajectory" to="$(var input/predicted_trajectory)"/>
    <remap from="input/twist" to="$(var input/twist)"/>
  </node>
</launch>
